cpu/atmega2560/periph/gpio.c

```diff
@@ -102,6 +102,24 @@ int gpio_init(gpio_t pin, gpio_dir_t dir, gpio_pp_t pullup)
     else {
         _SFR_MEM8(_ddr_addr(pin)) &= ~(1 << _pin_num(pin));
         res = bit_is_clear(_SFR_MEM8(_ddr_addr(pin)), _pin_num(pin));
+
+        if (res == 0) {
```

Comment:
"The if statement checks if the return value is equal 0 and that says, the bit is set."
Yes, if the bit is still **set** after we **cleared** it, it is clearly an error and we should return -1 *(and not bother with the pull up configuration)*.

Can you show me the code your calling the init with, as it works for me.